### PR TITLE
Document new environment variable expansion in projections

### PR DESCRIPTION
**Pull request by psakievich**
Adding docs and test for #42917

### CODE REVIEW

File: lib/spack/docs/environments.rst
Comment (alecbcs) [inline lib/spack/docs/environments.rst]:
```suggestion
Projections also permit environment and spack configuration variable
```

@psakievich this is awesome! Totally optional suggestion to clean up the language a little. 😃

**Review by alecbcs — DISMISSED** [review]
I'll defer to @alalazo or @tgamblin for a final review before merging but this looks great to me!

File: lib/spack/spack/test/projections.py
Comment (alalazo) [inline lib/spack/spack/test/projections.py]: Can you use `monkeypatch.setenv` here? Otherwise following tests will have a polluted environment.

File: lib/spack/spack/test/projections.py
Comment (alalazo) [inline lib/spack/spack/test/projections.py]: (Just OCD on my side) 
```suggestion
    """Test that env variables and spack config variables are expanded in projections"""
```

**Review by alalazo — CHANGES_REQUESTED** [review]
Thanks @psakievich Only a couple minor comments

**Comment by psakievich** [comment]
@spackbot fix style

File: lib/spack/spack/test/projections.py
Comment (alalazo) [inline lib/spack/spack/test/projections.py]:
```suggestion
```

**Review by alalazo — APPROVED** [review]

**Pull request merged**
